### PR TITLE
Motor subsystem base -> core

### PR DESCRIPTION
**Pull request by maya1414**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/motorsubsystem/MotorCommandsBuilder.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+public class MotorCommandsBuilder {
+
+	private final MotorSubsystem motorSubsystem;
+
+	public MotorCommandsBuilder(MotorSubsystem motorSubsystem) {
+		this.motorSubsystem = motorSubsystem;
+	}
+
+	public Command setPower(double power) {
+		return new RunCommand(() -> motorSubsystem.setPower(power), motorSubsystem).withName("set power");
```
Comment (hillelv) [inline src/main/java/frc/robot/subsystems/motorsubsystem/MotorCommandsBuilder.java:16]: consider adding the amount of power to the name

File: src/main/java/frc/robot/subsystems/motorsubsystem/MotorSubsystem.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
+package frc.robot.subsystems.motorsubsystem;
+
+import frc.robot.hardware.interfaces.ControllableMotor;
+import frc.robot.subsystems.GBSubsystem;
+import frc.utils.calibration.sysid.SysIdCalibrator;
+
+public class MotorSubsystem extends GBSubsystem {
+
+	ControllableMotor motor;
+	String logPath;
```
Comment (hillelv) [inline src/main/java/frc/robot/subsystems/motorsubsystem/MotorSubsystem.java:10]: GBSubsystem already has log path

File: src/main/java/frc/robot/subsystems/motorsubsystem/MotorSubsystem.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	MotorCommandsBuilder motorCommandsBuilder;
```
Comment (hillelv) [inline src/main/java/frc/robot/subsystems/motorsubsystem/MotorSubsystem.java:11]: why don't you initialize it?

File: src/main/java/frc/robot/subsystems/motorsubsystem/MotorSubsystem.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+	SysIdCalibrator sysIdCalibrator;
```
Comment (hillelv) [inline src/main/java/frc/robot/subsystems/motorsubsystem/MotorSubsystem.java:12]: I think they should be protected final

File: src/main/java/frc/robot/subsystems/motorsubsystem/MotorSubsystem.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+		motor.setPower(power);
+	}
+
+	public SysIdCalibrator getSysIdCalibrator() {
+		return sysIdCalibrator;
+	}
+
+	public String getLogPath() {
+		return super.getLogPath();
+	}
```
Comment (hillelv) [inline src/main/java/frc/robot/subsystems/motorsubsystem/MotorSubsystem.java:37]: why? no need to duplicate code

**Review by hillelv — CHANGES_REQUESTED** [review]
I think there should be a subsystem that uses IMotor (maybe abstract) and a class that inherits from it and uses ControllableMotor, there should be a way to create a non-controllable-motor subsystem

File: src/main/java/frc/robot/subsystems/motorsubsystem/MotorSubsystem.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
+package frc.robot.subsystems.motorsubsystem;
+
+import frc.robot.hardware.interfaces.ControllableMotor;
+import frc.robot.subsystems.GBSubsystem;
+import frc.utils.calibration.sysid.SysIdCalibrator;
+
+public class MotorSubsystem extends GBSubsystem {
+
+	ControllableMotor motor;
```
Comment (hillelv) [inline src/main/java/frc/robot/subsystems/motorsubsystem/MotorSubsystem.java:9]: you also didn't initialize this

File: src/main/java/frc/robot/subsystems/motorsubsystem/MotorSubsystem.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+	GBSubsystem logPath;
```
Comment (drorta) [inline src/main/java/frc/robot/subsystems/motorsubsystem/MotorSubsystem.java:10]: You don't need a GBSubsystem saved

File: src/main/java/frc/robot/subsystems/motorsubsystem/MotorSubsystem.java [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+	MotorCommandsBuilder motorCommandsBuilder;
+	protected final SysIdCalibrator sysIdCalibrator;
```
Comment (drorta) [inline src/main/java/frc/robot/subsystems/motorsubsystem/MotorSubsystem.java:12]: Consider making these private final

File: src/main/java/frc/robot/subsystems/motorsubsystem/MotorSubsystem.java [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+	public MotorSubsystem(GBSubsystem logPath, ControllableMotor motor) {
```
Comment (drorta) [inline src/main/java/frc/robot/subsystems/motorsubsystem/MotorSubsystem.java:14]:
```suggestion
	public MotorSubsystem(String logPath, ControllableMotor motor) {
```
Doesn't need to be GBSubsystem

File: src/main/java/frc/robot/subsystems/motorsubsystem/MotorSubsystem.java [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+		super(String.valueOf(logPath));
```
Comment (drorta) [inline src/main/java/frc/robot/subsystems/motorsubsystem/MotorSubsystem.java:15]:
```suggestion
		super(logPath);
```
You can just do this if it's a string, you don't need a GBSubsystem

File: src/main/java/frc/robot/subsystems/motorsubsystem/MotorSubsystem.java [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+
+import frc.robot.hardware.interfaces.ControllableMotor;
+import frc.robot.subsystems.GBSubsystem;
+import frc.utils.calibration.sysid.SysIdCalibrator;
+
+public class MotorSubsystem extends GBSubsystem {
+
+	ControllableMotor motor;
+	GBSubsystem logPath;
+	MotorCommandsBuilder motorCommandsBuilder;
```
Comment (drorta) [inline src/main/java/frc/robot/subsystems/motorsubsystem/MotorSubsystem.java:11]: you don't initialize this in the constructor

File: src/main/java/frc/robot/subsystems/motorsubsystem/MotorSubsystem.java [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+		motor.setPower(power);
+	}
+
+	public SysIdCalibrator getSysIdCalibrator() {
+		return sysIdCalibrator;
+	}
+
+	public String getLogPath() {
+		return String.valueOf(logPath);
+	}
```
Comment (drorta) [inline src/main/java/frc/robot/subsystems/motorsubsystem/MotorSubsystem.java:38]: You don't need this function, the class your inheriting from (GBSubsystem) already has this

**Review by drorta — CHANGES_REQUESTED** [review]
I like it, nice start! 🫡

**Pull request closed without merging**